osf/migrations/0145_pagecounter_data.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2018-11-12 17:18
+from __future__ import unicode_literals
+
+from django.db import migrations, connection
+
+
+def reverse_func(state, schema):
+    with connection.cursor() as cursor:
+        cursor.execute(
+            """
+            UPDATE osf_pagecounter
+            SET (action, guid_id, file_id, version) = ('download', NULL, NULL, NULL);
+            """
+        )
+
+def separate_pagecounter_id(state, schema):
+    """
+    Splits the data in pagecounter _id field of form action:guid_id:file_id:version into
+    four new columns: action(char), guid(fk), file(fk), version(int)
+    """
+    with connection.cursor() as cursor:
+        cursor.execute(
+            """
+            UPDATE osf_pagecounter PC
+            SET (action, guid_id, file_id, version) = (split_part(PC._id, ':', 1), G.id, F.id, NULLIF(split_part(PC._id, ':', 4), '')::int)
+            FROM osf_guid G, osf_basefilenode F
+            WHERE PC._id LIKE '%' || G._id || '%'
+            AND PC._id LIKE '%' || F._id || '%';
+            """
+        )
```

Comment:
Data migration to split the _id field into guid, file, version, and action fields.